tern/classes/image.py

```diff
@@ -16,6 +16,8 @@ class Image:
         manifest: the json object representing the image manifest
         config: the image config metadata
         layers: list of layer objects in the image
+        until_layer: image should be analyzed only up to this layer
+        layer_count: total number of layers in image.
```

Comment:
I don't think we need this -- we have the `layers` list property which contains all of the layer objects in the image. This is handy because if we want to find the total number of layers in the image we can use `len(layers)`

Reply:
len(layers) does not give us total number of layers in the image anymore as layers append is conditional. len(layers) gives us how many layers we need to work on currently.
layer_count is needed for the check in tern/analyze/default/container/run.py L78

I've checked the change against caching- I've made simple bash script to test it if when we populate only part of layers and in next scan we decide to scan full image- if all layers will be tested correctly. They were, the results were the same as the ones on master (except in different order but that differs each run). 

Sadly I've been working in VM, my host's disk had some issues, got mounted as -ro... after reboot & fsck vm didnt want to get up so i opted to delete it and create a new one. I can write it again and publish it as a comment in this PR

Reply:
When we do the layer loading in the beginning we get the total count of the layers in the image even when we don't analyze all of them. So `len(layers)` will be accurate as long as the image has been loaded (even if we don't analyze all of them).

Reply:
Ah, I see your commits made edits to only load certain layers. Let me look a little more at this.

Reply:
Here's where I am at right now thinking about this issue: I'm not sure loading only the layers we need gives us much advantage and the trade off is passing more information from command line args to objects and then keeping track of that information and adding more logic around it. Especially if the only reason we are doing it this way is to avoid the prospector errors. I can get rid of the prospector errors and it has been on my radar to do this. Most of the time, the largest layer to load is going to be the base os layer which we will always need to load so I don't think we are saving much time by only loading up until certain layers. To me it makes more sense to instead limit the analysis depending on the layers that we need if we are trying to save processing time with the `--layer` option. This `--layers` option is really about the output report which is why I am hesitant to change a core functionality like loading for it when it would be much easier to just change the info that gets printed or limit analysis. Thoughts? Again, apologies for missing the discussion on this when you opened your WIP PR.

Reply:
I mean loading only part of image makes sense to me - but its true that not that much time is saved by it.
That is just what I'd expect of the app when asked to analyse only part of the image. 
I dont know the whole codebase but it seems pretty straightforward considering most of the code I've seen so far iterates through image.layers thus making this feature simple.
If you see scenarios where it could be too much of maintenance burden I can try to revert it back (though I'll have to check how to pull force-pushes from github PR or go through git reflog) Other idea I  can think of is adding some tests- but I'd need some help on that (what parts should be in the scope of such tests + I'm not sure if I'll manage to write those tests this/next week)

Reply:
Hm, ok... maybe `total_layers` is a better variable name here?